src/lib/services/task_results.ts

```diff
@@ -25,6 +27,70 @@ export async function getTaskResults(userId: string): Promise<TaskResults> {
   return await relateTasksAndAnswers(userId, tasks, answers);
 }
 
+export async function cpoyTaskResults(
+  sourceUserName: string,
+  destinationUserName: string,
+): Promise<Check[]> {
+  const checks = [];
+  const failureMessage = { label: 'コピーが失敗しました。', status: false };
+  const sourceUser = await getUser(sourceUserName);
+  const destinationUser = await getUser(destinationUserName);
+  if (sourceUser == null) {
```

Comment:
[Recommended] 私のワーキングメモリが小さいだけかもしれませんが、38-72行目の処理は4つの条件分岐の内容が類似しており、やや冗長に見える気がします。ユーザと回答を検証する関数をそれぞれ切り出した方が全体の見通しや単体テストがしやすくなると思いますが、いかがでしょうか?

```suggestion
// 元の関数
    const sourceUserExists = isExistingUser(sourceUserName, checks, failureMessage);

  if (!sourceUserExists) {
    return checks;
  }
  
// 切り出したメソッドの一例（もっとスマートな方法があるかもしれません）
function isExistingUser(userName: string, messages: { label: string; status: boolean }[], failureMessage: { label: string; status: boolean }) {
  if (userName == null) {
    messages.push({
      label: `${userName} が存在しません。コピーを中止します。`,
      status: false,
    });
    messages.push(failureMessage);
    return false;
  } else {
    messages.push({ label: `${userName} が存在することを確認しました`, status: true });
    return true;
  }
} 
```

Reply:
ユーザを検証する関数を外に出す　https://github.com/AtCoder-NoviSteps/AtCoderNoviSteps/pull/1371/commits/e33e470523f15023b18f661d6e708c666020160f